new-lamassu-admin/src/pages/Authentication/Reset2FA.js

```diff
@@ -0,0 +1,201 @@
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import { makeStyles, Grid } from '@material-ui/core'
+import Paper from '@material-ui/core/Paper'
+import gql from 'graphql-tag'
+import QRCode from 'qrcode.react'
+import React, { useReducer, useState } from 'react'
+import { useLocation, useHistory } from 'react-router-dom'
+
+import { ActionButton, Button } from 'src/components/buttons'
+import { CodeInput } from 'src/components/inputs/base'
+import { H2, Label2, Label3, P } from 'src/components/typography'
+import { ReactComponent as Logo } from 'src/styling/icons/menu/logo.svg'
+import { primaryColor } from 'src/styling/variables'
+
+import styles from './shared.styles'
+
+const QueryParams = () => new URLSearchParams(useLocation().search)
```

Comment:
I think is better to have this inside the component function, we can't really access useLocation from elsewhere.